runtime/src/main/java/org/corfudb/protocols/wireprotocol/ClientHandshakeHandler.java

```diff
@@ -129,11 +138,12 @@ public void channelRead(ChannelHandlerContext ctx, Object m)
         }
 
         log.info("channelRead: Handshake succeeded. Server Corfu Version: [{}]", corfuVersion);
-        log.debug("channelRead: There are [{}] messages in queue to be flushed.", this.messages.size());
-        // Flush messages in queue
-        while (!messages.isEmpty()) {
-            ctx.writeAndFlush(messages.poll());
-        }
+
+        // Flush messages in backlog
+        messages.forEach(ctx::writeAndFlush);
+        messages.clear();
+        requestMessages.forEach(ctx::writeAndFlush);
+        requestMessages.clear();
```

Comment:
same here

Reply:
Note: after internal discussion, this will be addressed in a separate PR.